qdlt/qdltmsg.cpp

```diff
@@ -628,3 +628,89 @@ QString QDltMsg::toStringPayload() const
 
     return text;
 }
+
+void QDltMsg::genMsg()
+{
+    DltStandardHeader standardheader;
+    DltStandardHeaderExtra headerextra;
+    DltExtendedHeader extendedheader;
+    QDltArgument argument;
+
+    // clear existing payload
+    payload.clear();
+
+    // Generate payload for all arguments
+    for(int num=0;num<arguments.size();num++) {
+        if(getArgument(num,argument)) {
+            argument.getArgument(payload,true);
+        }
+
+    }
+
+    // set payload size
+    payloadSize = payload.size();
+
+    // clear existing header
+    header.clear();
+
+    // write standardheader
+    standardheader.htyp = 0x01 << 5; /* intialise with version number 0x1 */
+    if(endianness == DltEndiannessBigEndian) {
+        standardheader.htyp |= DLT_HTYP_MSBF;
+    }
+    if(mode == DltModeVerbose) {
+        standardheader.len = DLT_SWAP_16(sizeof(DltStandardHeader) + sizeof(DltExtendedHeader) + payload.size());
```

Comment:
Do you really want to SWAP or move to a specific endianess? -> DLT_HTOLE/BE_16 ?

Reply:
DLT Viewer only runs on little endian, need to swap always for DLT Header, as it is in big endian.

Reply:
Changed to DLT_HTOBE_16